### ISSUE TITLE
Feature request: Button to disable Hot-Reload temporarly

### ISSUE THREAD

**Issue by plehnen** [cypress-io/cypress#4611]
### Current behavior:

In dev mode, when I edit my code and unfocus the IDE the file gets auto saved and the cypress test runs again automatically. That is overall a great feature.
But a frequent usecase is to copy stuff from the current result into the IDE and while doing that, the page reloads and one has to wait until it finishs, scroll to the desired position again, copy the line, insert it into the IDE... and for the next all this happens again.

### Desired behavior:

So it would be very nice to have another button on top (e.g. next to the "Disable Auto-Scrolling") to be able to temorarly disable this hot-reload auto-re-run test behaviour.

### Steps to reproduce: (app code and test code)

Open cypress in dev mode, run a test, edit the test and save the test file.

### Versions
3.3.1

**Comment by jennifer-shehane** [comment]
Thanks for opening this feature request.

You can set currently `watchForFileChanges: false` from within your Cypress configuration to turn off watching spec files altogether. https://on.cypress.io/configuration#Global

You may also want to consider changing your IDE settings so that it does not 'auto-save' on change or so that there is a delay on 'auto-save' so that files are not saved until you manually save. For example, settings in VS Code for this: https://code.visualstudio.com/docs/editor/codebasics#_save-auto-save

Adding a button to disable watching spec files is not something we see as within our scope of feature work, so we will be closing this issue. We will reconsider and reopen if we see this issue gains more 👍 or comments in support.

**Comment by cmolenda** [comment]
I would have use for this as well. My situation is this.

1. I'm writing specs for new feature work.
2. One of the specs fails after it had previously passed but without me changing related code (sign of a flakey spec).
3. I want to be able to adjust the spec code and save as I'm doing so without losing the information from the previous run. This is particularly useful for fixing a flaky/flapping spec, since by its nature I probably can't readily reproduce it. This makes a caught failure much more valuable and much worse to lose.

**Comment by colinmollenhour** [comment]
In my case in addition to the above reason I need to first reload a database snapshot before running some of my tests. If I use `watchForFileChanges: false` then it doesn't reload right away which is good, but then there is no way to trigger a reload and I have to kill the entire IDE and boot it up again which is decidedly slower.

**Comment by raresserban** [comment]
A button besides the `Disable Auto-scrolling` and `Run All Tests` button could be added to the Cypress UI to temporarily disable auto refresh. I really don't know why this isn't a feature already considering most IDEs auto save and people use the Cypress rendered website as reference while writing the tests themselves...

**Comment by mP1** [comment]
I know this has been closed, but this feature would be especially useful for the following use case.

Imagine adding several new tests, some pass and some fail. In this case I would love to pass auto reload, so i can examine each of the failed tests one by one and update the source one by one. Problem is with the auto reload, as soon as I make changes to address the first fail, the tests are re-run and the other failed tests are lost.

I appreciate I could do the suggestions above, but its far more convenient to have control of pausing auto reload by toggling a button.

In my case, i could fix each of the failed tests, then turn autoreload back on and let the tests run again, until success.

**Comment by jlaustill** [comment]
This feature would be extremely useful. It isn't a great workflow to disable auto-saving when you use it for everything else and constantly have to switch it. It would be great to be able to hit pause hot reload and go back and forth between a page inspection finding elements and adding them to your test without the test starting over each time.  I would be happy to help contribute to this feature at some point.

**Comment by shishir-dwivedi-ethos** [comment]
I totally agree with the suggestion given above, I recently switched from Selenium to Cypress and found hot reload creates a problem while fixing a test case. A simple button on UI or any cypress flag would be helpful to freeze the current state so that we can refer to the failure, fix the script and run them again. This feature would be extremely useful.

**Comment by Wilt** [comment]
Seems this got lots of comments and thumbs up, please consider scheduling this feature for some future release? I understand it may not have highest priority for the development team, but there clearly seems to be interest from the community.

**Comment by nt294** [comment]
Is this something that was ever added?
